### PR TITLE
Fix observation deletion crash

### PR DESCRIPTION
**Pull request by Chrischuck**
#545 

I'm guessing what happened was that the response from the server was cached and thus was searching for an ID in realm that didn't exist.

I wasn't able to repro the issue after this, but please lmk if you can.

### CODE REVIEW

**Comment by albullington** [comment]
Hi @Chrischuck, I can still reproduce this issue (using a real iPhone device). I created a new observation in the app, didn't upload to the server, tapped the observation, deleted the photo, returned to the observations list screen, and waited.

After maybe 2 minutes of waiting, I get the following error message: 
IMG_38066C7B5B0C-1

**Comment by Chrischuck** [comment]
@albullington i see, i only tested deleting an uploaded obs, i can check out that test case

**Comment by albullington** [comment]
@Chrischuck did this resolve the error for you? I'm still seeing the error on iOS, and it seems to appear when I scroll down in the list. The deleted observation isn't being removed from MyObservations list, which might be part of the problem.

On Android, I'm seeing a slightly different, but related error:
Screenshot_20230329-110832

File: src/sharedHooks/useLocalObservations.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ const useLocalObservations = ( ): Object => {
 
       if ( allObsToUpload.length < unsyncedObs.length ) {
```
[the remark this answers is not in the document]
Comment (kueda) [reply src/sharedHooks/useLocalObservations.js:36]: Seems like it will probably be fine.

**Comment by Chrischuck** [comment]
@albullington Yep, it resolves the issue for me and gets removed from myobservations list. Although I have to use a simulator/emulator


https://user-images.githubusercontent.com/13311268/228645921-26a58708-ef42-460b-a781-049dc5365bba.mov

**Review by kueda — APPROVED** [review]
I can't recreate the error using the repro steps in the issue, and it seems to work ok in Android and iOS for me.

**Pull request merged**